src/backend/core/services/lobby_service.py

```diff
@@ -0,0 +1,271 @@
+"""Wip."""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, List, Optional, Tuple
+
+from django.conf import settings
+from django.core.cache import cache
+
+import aiohttp
+from asgiref.sync import async_to_sync
+from livekit import api as livekit_api
+from livekit.api import room_service as room_service_api
+
+from core import utils
+
+logger = logging.getLogger(__name__)
+
+
+class LobbyParticipantStatus(Enum):
+    """Possible states of a participant in the lobby system.
+    Values are lowercase strings for consistent serialization and API responses.
+    """
+
+    UNKNOWN = "unknown"
+    WAITING = "waiting"
+    ACCEPTED = "accepted"
+    DENIED = "denied"
+
+
+class LobbyError(Exception):
+    """Base exception for lobby-related errors."""
+
+
+class LobbyParticipantParsingError(LobbyError):
+    """Raised when participant data parsing fails."""
+
+
+class LobbyParticipantNotFound(LobbyError):
+    """Raised when participant is not found."""
+
+
+class LobbyNotificationError(LobbyError):
+    """Raised when LiveKit notification fails."""
+
+
+@dataclass
+class LobbyParticipant:
+    """Wip"""
+
+    status: LobbyParticipantStatus
+    username: str
+    id: str
+
+    def to_dict(self) -> Dict[str, str]:
+        """Wip"""
+        return {"status": self.status.value, "username": self.username, "id": self.id}
+
+    @classmethod
+    def from_dict(cls, data: dict) -> "LobbyParticipant":
+        """Wip"""
+        try:
+            status = LobbyParticipantStatus(
+                data.get("status", LobbyParticipantStatus.UNKNOWN.value)
+            )
+            return cls(status=status, username=data["username"], id=data["id"])
+        except (KeyError, ValueError) as e:
+            logger.exception("Error creating Participant from dict:")
+            raise LobbyParticipantParsingError("Invalid participant data") from e
+
+
+class LobbyService:
+    """Service for managing participant access through a lobby system.
+
+    Handles participant entry requests, status management, and notifications
+    using Redis cache for state management and LiveKit for real-time updates.
+    """
+
+    @staticmethod
+    def _get_cache_key(room_id: str, participant_key: str) -> str:
+        """Generate cache key for participant(s) data."""
+        return f"{settings.LOBBY_KEY_PREFIX}_{room_id}_{participant_key}"
```

Comment:
```suggestion
        return f"{settings.LOBBY_KEY_PREFIX}_{room_id!s}_{participant_key!s}"
```
`!` is the conversion mark in a fstring and `s` is for string. Doing this you don't have to manage the casting in caller methods, don't need anymore to call `str(room.id)` in the viewset for example.